tests_consume/test_via_rlp.py

```diff
@@ -0,0 +1,209 @@
+
+@pytest.fixture(scope="function")
+def to_geth_genesis(test_case: TestCase, test_case_fixture: Fixture) -> dict:
+    """
+    Convert the genesis block header of the current test fixture to a geth genesis block.
+    """
+    geth_genesis = {
+        "nonce": test_case_fixture.genesis["nonce"],
+        "timestamp": test_case_fixture.genesis["timestamp"],
+        "extraData": test_case_fixture.genesis["extraData"],
+        "gasLimit": test_case_fixture.genesis["gasLimit"],
+        "difficulty": test_case_fixture.genesis["difficulty"],
+        "mixhash": test_case_fixture.genesis["mixHash"],
+        "coinbase": test_case_fixture.genesis["coinbase"],
+        # TODO: retrieve pre_state from the fixture? Instead of the json
+        # (and potentially remove the json_as_dict field completely from TestCase)
+        "alloc": test_case.json_as_dict["pre"],
+    }
+    # TODO: Use ethereum_test_forks to detect new fields automatically?
+    for field in ["baseFeePerGas", "withdrawalsRoot", "blobFeePerGas", "blobGasUsed"]:
+        if field in test_case_fixture.genesis:
+            geth_genesis[field] = test_case_fixture.genesis[field]
+    return geth_genesis
+
+
+@pytest.fixture
+def buffered_genesis(to_geth_genesis: dict) -> io.BufferedReader:
+    genesis_json = json.dumps(to_geth_genesis)
+    genesis_bytes = genesis_json.encode("utf-8")
+    return io.BufferedReader(io.BytesIO(genesis_bytes))
+
+
+@pytest.fixture
+def buffered_blocks_rlp(blocks_rlp: List[str], start=1) -> List[io.BufferedReader]:
+    block_rlp_files = []
+    for i, block_rlp in enumerate(blocks_rlp):
+        blocks_rlp_bytes = bytes.fromhex(block_rlp[2:])
+        blocks_rlp_stream = io.BytesIO(blocks_rlp_bytes)
+        block_rlp_files.append(io.BufferedReader(blocks_rlp_stream))
+    return block_rlp_files
+
+
+@pytest.fixture
+def files(
+    test_case: TestCase,
+    client_type: ClientType,
+    buffered_genesis: io.BufferedReader,
+    buffered_blocks_rlp: list[io.BufferedReader],
+) -> Mapping[str, io.BufferedReader]:
+    """
+    Define the files that hive will start the client with.
+
+    The files are specified as a dictionary whose:
+    - Keys are the target file paths in the client's docker container, and,
+    - Values are in-memory buffered file objects.
+    """
+    files = {f"/blocks/{i:04d}.rlp": block_rlp for i, block_rlp in enumerate(buffered_blocks_rlp)}
+    if client_type.name == "nethermind":
+        files["/chainspec/test.json"] = buffered_genesis
+    else:
+        files["/genesis.json"] = buffered_genesis
+    return files
+
+
+@pytest.fixture
+def environment(test_case: TestCase) -> dict:
+    env = {
+        "HIVE_FORK_DAO_VOTE": "1",
+        "HIVE_CHAIN_ID": "1",
+    }
+    assert test_case.fixture.fork in ruleset, "Oops, should never get here"
+    for k, v in ruleset[test_case.fixture.fork].items():
+        env[k] = f"{v:d}"
+    if test_case.fixture.seal_engine == "NoProof":
+        env["HIVE_SKIP_POW"] = "1"
+    return env
```

Comment:
I think we can remove the PoW check here, as I'm pretty sure we don't need to use `HIVE_SKIP_POW` within hive any more. I actually don't think we even need to add `sealEngine: "NoProof"` to our fixtures as well :D

Suggestion below. I'd still really like to change env prefix from `HIVE_FORK_DAO_VOTE` to `CLIENT_FORK_DAO_VOTE`, for client specific environment variables. Although I do admit having `HIVE_` makes the abstraction clearer within EEST 🤙

```suggestion
@pytest.fixture
def client_environment(test_case: TestCase) -> dict:
    """
    Defines the environment that hive will start the client with
    using the fork rules specific for the simulator.
    """
    client_env = {
        "HIVE_FORK_DAO_VOTE": "1",
        "HIVE_CHAIN_ID": "1",
        "HIVE_NODETYPE": "full",
        **{k: f"{v:d}" for k, v in forks.get(test_case.fixture.fork, {}).items()},
    }
    return client_env
```

Reply:
Note I had issues with clients in the engine (pyspec) sim in the past hence specifying that the client should be a full node.

I wondered if we would get a could get a speed up setting `"HIVE_NODETYPE": "snap",` but it had no noteable difference. 

With regards to speed ups: using `xdist` alongside copying the pre-loaded docker-ized client would be amazing. If we can find a way within `hive` or `hive.py` to first load the built client image to the point before a test is consumed, we can keeping copying this container over and over. If we have `>10000` test cases the speed up will be worth it.

Reply:
Thanks for these! :pray: 

Added your suggestions to the `environment` fixture in a3836e96. Nice dictionary comprehension!

Made a note on copying the pre-loaded dockerized client - I'd forgotten about that!